### ISSUE TITLE
DataSources don't show BoneCPDatasource (BoneCP 0.8.0.RELEASE)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [psi-probe/psi-probe#405]
```
Hi,

The BoneCPDatasourceAccessor doesn't work with the newest BoneCP Release 
(0.8.0-RELEASE). There is a new Wrapper class (FinalWrapper) in 
BoneCPDataSource, so the cast to BoneCP is broken:

final Field poolField = BoneCPDataSource.class.getDeclaredField("pool");
poolField.setAccessible(true);
final BoneCP pool = (BoneCP) poolField.get(source);

The Exception is: java.lang.ClassCastException: 
com.jolbox.bonecp.BoneCPDataSource$FinalWrapper cannot be cast to 
com.jolbox.bonecp.BoneCP

There is a new getter in BoneCPDataSource to get the Pool directly, so 

final BoneCP pool = source.getPool();

should do the work know.

I'm using probe version 2.3.3 on Tomcat 7.0.42 with jdk 1.7.0_25

Kind regards
Jan-Hendrik
```

Original issue reported on code.google.com by `socialme...@albamoda.de` on 24 Apr 2014 at 11:58

**Comment by jander99** [comment]
Scanning Milestone-2.4.0.  

Is this attachment good to go to include in master?

**Comment by MALfunction84** [comment]
@jander99 I don't know. Didn't have a test environment for it.

**Comment by diogosantana** [comment]
Vote to move from 2.4 to 3.0 or future.
With my fix to #466 we can focus on #394 to close version 2.4.
